Tools/baselineDef.cc

```diff
@@ -1264,6 +1366,153 @@ bool BaselineVessel::GetMHT() const
+bool BaselineVessel::photonPassEtaPtCuts(const TLorentzVector& photon) const
+{
+    const double maxEta = 2.5;
+    const double minPt = 200.0;
+    const double photonEta = photon.Eta();
+    const double photonPt  = photon.Pt();
+    return (fabs(photonEta) < maxEta && photonPt > minPt);
+}
+
+
+
+// ===  FUNCTION  ============================================================
+//         Name:  BaselineVessel::GetPhotons
+//  Description:  Get photons passing eta, pt, and ID selection 
+// ===========================================================================
+bool BaselineVessel::GetPhotons() const
+{
+    // varialbes from ntuples
+    const auto& gammaLVec           = tr->getVec<TLorentzVector>("gammaLVec");     // reco photons
+    const auto& gammaLVecGen        = tr->getVec<TLorentzVector>("gammaLVecGen");  // gen photon
+    const auto& loosePhotonID       = tr->getVec<unsigned int>("loosePhotonID");   // loose ID
+    const auto& mediumPhotonID      = tr->getVec<unsigned int>("mediumPhotonID");  // medium ID
+    const auto& tightPhotonID       = tr->getVec<unsigned int>("tightPhotonID");   // tight ID
+    const auto& genMatched          = tr->getVec<float>("genMatched");            // matched to gen photon
+    const auto& met                 = tr->getVar<float>("met");
+    const auto& metphi              = tr->getVar<float>("metphi");
+    
+    // new variables
+    auto* gammaLVecReco             = new std::vector<TLorentzVector>(); // reco photon
+    auto* gammaLVecRecoEta          = new std::vector<TLorentzVector>(); // reco photon passing eta cuts
+    auto* gammaLVecPassLooseID      = new std::vector<TLorentzVector>(); // reco photon passing loose ID 
+    auto* gammaLVecPassMediumID     = new std::vector<TLorentzVector>(); // reco photon passing medium ID 
+    auto* gammaLVecPassTightID      = new std::vector<TLorentzVector>(); // reco photon passing tight ID 
+    //auto* metLVec                   = new TLorentzVector(); // already defined in baseline
+    auto* metWithPhotonLVec         = new TLorentzVector();
+
+    // toggle debugging print statements
+    bool debug = false;
+
+    //variables to be used in the analysis code
+    double metWithPhoton = -999.9;
+    double metphiWithPhoton = -999.9;
+    bool passPhotonSelection = false;
+        
+    //Select reco photons; only eta cuts for now
+    for(int i = 0; i < gammaLVec.size(); ++i)
+    {
+        gammaLVecReco->push_back(gammaLVec[i]);
+      // passing ECAL barrel/endcap eta cuts
+      // this needs to be done prior to any other cuts (pt, gen matched, etc)
+      // this cut should match passAcc which is done in StopTupleMaker/SkimsAUX/plugins/PhotonIDisoProducer.cc
+      if (photonPassECALEtaCuts(gammaLVec[i])) 
+      {
+        gammaLVecRecoEta->push_back(gammaLVec[i]);
+      }
+    }
+    
+    // check vector lengths: gammaLVecRecoEta should have the same length as photon ntuple values for which passAcc=true
```

Comment:
This section is confusing and can be simplified, why do we even need gammaLVecReco or gammaLVecRecoEta ?